src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -223,13 +223,11 @@ public DetectBoot(DetectBootFactory detectBootFactory, DetectArgumentState detec
     }
 
     private void publishCollectedPropertyValues(PropertyConfiguration propertyConfiguration, EventSystem eventSystem) throws IllegalAccessException {
-        PropertyInfoCollector propertyInfoCollector = new PropertyInfoCollector(propertyConfiguration);
-        List<PropertyInfo> propertyInfo = propertyInfoCollector.collectPropertyInfo(DetectProperties.allProperties(), PropertyInfoCollector.maskPasswordsAndTokensPredicate());
-        Map<String, String> propertyValues = new HashMap<>();
-        propertyInfo.forEach(
-            it -> propertyValues.put(it.getKey(), it.getValue())
-        );
-        eventSystem.publishEvent(Event.PropertyValuesCollected, new PropertyValues(propertyValues));
+        Map<String, String> rawPropertyKeyValues = propertyConfiguration.getRawKeyValueMap(new HashSet<>(DetectProperties.allProperties()));
+        PropertyMasker propertyMasker = new PropertyMasker();
+        Predicate<String> shouldMaskRawValue = propertyKey -> propertyKey.toLowerCase().contains("password") || propertyKey.toLowerCase().contains("api.token") || propertyKey.toLowerCase().contains("access.token");
+        Map<String, String> maskedRawPropertyKeyValues = propertyMasker.maskRawValues(rawPropertyKeyValues, shouldMaskRawValue);
```

Comment:
And i do find it strange we compute the masks twice instead of just sharing the computation.

Reply:
I agree it would be ideal to have it computed once, I'm just not sure where it would make sense for the computation to live.  The interaction between the subprojects makes it tricky.

Reply:
Refactored so DetectBoot is the sole computer.